### PR TITLE
Compatiblity with Symfony 4.3

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: features/jsonld/context.feature [hunk 1]
```diff
@@ -41,7 +41,6 @@ Feature: JSON-LD contexts generation
             "jsonData": "Dummy/jsonData",
             "arrayData": "Dummy/arrayData",
             "nameConverted": "Dummy/nameConverted",
-            "id": "Dummy/id",
```
Comment (dunglas) [inline features/jsonld/context.feature]: Why do we need to change this @soyuka? It looks like a BC break.

Comment (alanpoulain) [reply features/jsonld/context.feature]: I've done this change. It's because of you :stuck_out_tongue:
https://github.com/symfony/symfony/pull/31486

Comment (alanpoulain) [reply features/jsonld/context.feature]: That's also why I've added what you've done for MongoDB too: https://github.com/api-platform/core/pull/2784/files#diff-c5fed7f9ec0eb28b0039798760fac3ac

Comment (dunglas) [reply features/jsonld/context.feature]: Got it!

**Comment by dunglas** [comment]
Thank you very much @soyuka and @alanpoulain for the help!

File: composer.json [hunk 2]
```diff
@@ -71,10 +71,10 @@
         "symfony/expression-language": "^3.4 || ^4.0",
         "symfony/finder": "^3.4 || ^4.0",
         "symfony/form": "^3.4 || ^4.0",
-        "symfony/framework-bundle": "^4.2",
+        "symfony/framework-bundle": "^4.3",
         "symfony/mercure-bundle": "*",
-        "symfony/messenger": "^4.2",
-        "symfony/phpunit-bridge": "^4.3",
+        "symfony/messenger": "^4.3",
```
Comment (teohhanhui) [inline composer.json]: Is it a good idea to bump the dependencies in a patch version? It means that our (potential) future security fixes will not reach the users who are stuck on `<=2.4.4`

Comment (teohhanhui) [reply composer.json]: What this essentially means:

You must upgrade Symfony Messenger to continue to receive security fixes for 2.4

File: src/Serializer/ItemNormalizer.php [hunk 3]
```diff
@@ -30,8 +30,6 @@
 /**
  * Generic item normalizer.
  *
- * @final
```
Comment (teohhanhui) [inline src/Serializer/ItemNormalizer.php]: I don't think this is right. The GraphQL ItemNormalizer should stop extending this class instead.

File: tests/Fixtures/app/AppKernel.php [hunk 4]
```diff
@@ -96,10 +97,11 @@ protected function configureContainer(ContainerBuilder $c, LoaderInterface $load
 
         $loader->load(__DIR__."/config/config_{$this->getEnvironment()}.yml");
 
+        $alg = class_exists(SodiumPasswordEncoder::class) && SodiumPasswordEncoder::isSupported() ? 'auto' : 'bcrypt';
```
Comment (teohhanhui) [inline tests/Fixtures/app/AppKernel.php]: Uhh... Symfony really screwed this one up. :x

**Comment by teohhanhui** [comment]
It seems like this is not addressed? https://github.com/api-platform/core/issues/2856

File: composer.json (hunk 2 above)
Comment (alanpoulain) [reply composer.json]: It's not for a patch version but a minor one. I think it's OK because this component is experimental in Symfony anyway.

Comment (teohhanhui) [reply composer.json]: Uhh... Didn't notice this was targeting `master` 🙈

**Comment by soyuka** [comment]
I'm not sure to understand how #2856 is an issue here...

**Comment by teohhanhui** [comment]
@soyuka No, it's not an issue here. Just that I'd consider #2856 essential for Symfony 4.3 compatibility. Because it's not doing the right thing now.

File: tests/Fixtures/app/AppKernel.php (hunk 4 above)
Comment (maxhelias) [reply tests/Fixtures/app/AppKernel.php]: It's not bcrypt if sodium isn't supported but the native encoder. I do not know if it's important here.
The bcrypt encoder is now deprecated

Comment (dunglas) [reply tests/Fixtures/app/AppKernel.php]: It's only for old versions so I think we don't care (this code will go be thrown away at some point).

**Comment by soyuka** [comment]
True although I'm not sure to get what we need to do on our side.

**Comment by bastnic** [comment]
Why only master and not 2.4 branch ? My tests are full of `  59403x: Method "ApiPlatform\Core\Serializer\ItemNormalizer::createChildContext()" will have a third "?string $format" argument in version 5.0; not defining it is deprecated since Symfony 4.3.`

**Comment by dunglas** [comment]
Because Messenger. Messenger 4.3 introduces some BC breaks, that in cascade force us to add some BC breaks too. Also, to use Messenger 4.3 we had to bump the version of several other Symfony components. We then decided that it wasn't a good idea to introduce such changes in a patch release.
Anyway, we're working hard on finishing 2.5, a beta should be available soon.

**Comment by teohhanhui** [comment]
We could certainly backport the other changes that are not related to Symfony Messenger.

**Comment by dunglas** [comment]
Maybe some of them yes! Feel free to open a PR!

File: src/Bridge/Symfony/Messenger/DataPersister.php [hunk 5]
```diff
@@ -88,6 +88,6 @@ public function persist($data, array $context = [])
      */
     public function remove($data, array $context = [])
     {
-        $this->messageBus->dispatch(new Envelope($data, new RemoveStamp()));
+        $this->messageBus->dispatch(new Envelope($data, [new RemoveStamp()]));
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Messenger/DataPersister.php]:
```php
$this->messageBus->dispatch(
    (new Envelope($data))
        ->with(new RemoveStamp())
);
```

Comment (teohhanhui) [reply src/Bridge/Symfony/Messenger/DataPersister.php]: I'm not sure why it's still possible to pass in the stamps through the constructor...

File: features/graphql/mutation.feature [hunk 6]
```diff
@@ -324,7 +324,7 @@ Feature: GraphQL mutation support
     When I send the following GraphQL request:
     """
     mutation {
-      createDummy(input: {_id: 12, name: "", foo: [], clientMutationId: "myId"}) {
```
Comment (teohhanhui) [inline features/graphql/mutation.feature]: @dunglas @alanpoulain Do you know what's going on here?

**Pull request merged**